File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) 
    {
        int maxSum = nums[0];
        int currMax = nums[0];

        int n = nums.size();

        for(int i= 1; i < n ;i++)
        {
            currMax = max(nums[i],currMax + nums[i]);
            maxSum = max(maxSum,currMax);
        }

        return maxSum;
        
    }
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int crSum = nums[0];

        int n = nums.length;

        for(int i=1; i < n ;i++)
        {
            crSum = Math.max(crSum + nums[i] , nums[i]);
            maxSum = Math.max(maxSum, crSum);
        }

        return maxSum;
        
    }
}